cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -2203,6 +2206,53 @@ if(ARROW_WITH_RAPIDJSON)
   endif()
 endif()
 
+
+macro(build_qpl)
+  message(STATUS "Building QPL from source")
+  set(QPL_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/qpl_ep/src/qpl_ep-install")
+  if(MSVC)
+      set(QPL_STATIC_LIB_NAME qplstatic.lib)
+  else()
+    set(QPL_STATIC_LIB_NAME libqpl.a)
+  endif()
+  set(QPL_STATIC_LIB "${QPL_PREFIX}/lib64/${QPL_STATIC_LIB_NAME}")
+  set(QPL_CMAKE_ARGS
+      ${EP_COMMON_CMAKE_ARGS}
+      -DCMAKE_BUILD_TYPE=Release
+      "-DCMAKE_INSTALL_PREFIX=${QPL_PREFIX}"
+      EXCLUDE_FROM_ALL NOT)                      
+
+
+  ExternalProject_Add(qpl_ep
+      ${EP_LOG_OPTIONS}
+      GIT_REPOSITORY    https://github.com/intel/qpl.git
+      GIT_TAG           v0.2.1
+      GIT_SUBMODULES_RECURSE TRUE
```

Comment:
We want to use source archive instead of Git repository to reduce Git dependency on build.
See also `cpp/thirdparty/versions.txt`.

Reply:
There is a known issue on intel/qpl: 
QPL cannot be built from direct downloadable files (.tar, .tgz) since it has submodules which are not included to the archives by GitHub during release creation(https://github.com/intel/qpl/releases). This will be solved on next version.
For now, can we just use git repository to pull released code? Maybe we can create an issue/Jira to track this problem if the code is merged.

Reply:
When the next release is scheduled?

BTW, we can apply a patch with `externalproject_add()`: https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L4104-L4113

Reply:
> When the next release is scheduled?

It would be fixed in Q1, 2023